frontend/src/metabase/components/form/FormError/FormError.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { ReactNode, useEffect, useMemo, useRef } from "react";
+import { getErrorMessageWithBoldFields } from "metabase/lib/form";
+import { BaseFieldDefinition } from "metabase-types/forms";
+
+import { ErrorCard, ScrollAnchor, WarningIcon } from "./FormError.styled";
+
+interface FormErrorProps {
+  error?: string;
+  formFields?: BaseFieldDefinition[];
+  anchorMarginTop?: number;
+}
+
+export default function FormError({
+  error,
+  formFields,
+  anchorMarginTop,
+}: FormErrorProps) {
+  const scrollAnchorRef = useRef<HTMLDivElement>(null);
+  useEffect(() => {
+    if (error) {
+      scrollAnchorRef.current?.scrollIntoView();
+    }
+  }, [error]);
+
+  const renderedError = useMemo(
+    () => getErrorMessageWithBoldFields(error, formFields),
+    [error, formFields],
+  );
```

Comment:
Memoize `renderedError` to prevent this component from re-rendering every time other props changes as `getErrorMessageWithBoldFields` produces React element.